react/features/video-quality/components/VideoQualityLabel.native.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { translate } from '../../base/i18n';
+import { CircularLabel } from '../../base/label';
+import { normalizeStyleProps } from '../../base/styles';
+
+import AbstractVideoQualityLabel, {
+    _abstractMapStateToProps,
+    type Props as AbstractVideoQualityLabelProps
+} from './AbstractVideoQualityLabel';
+import styles from './styles';
+
+type Props = {
+    ...$Exact<AbstractVideoQualityLabelProps>,
+
+    /**
+     * Style of the component passed as props.
+     */
+    style: ?Object,
+
+    /**
+     * Calculated style of the props based on the video quality.
+     */
+    _style: ?Object
+};
+
+/**
+ * React {@code Component} responsible for displaying a label that indicates
+ * the displayed video state of the current conference.
+ *
+ * NOTE: Due to the lack of actual video quality information on mobile side,
+ * this component currently only displays audio only indicator, but the naming
+ * is kept consistent with web and in the future we may introduce the required
+ * api and extend this component with actual quality indication.
+ */
+class VideoQualityLabel extends AbstractVideoQualityLabel<Props> {
+
+    /**
+     * Implements React {@link Component}'s render.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { style, t, _style } = this.props;
+
+        if (typeof _style === 'undefined') {
+            // We don't have info about the quality so no need for the indicator
+            return null;
+        }
+
+        return (
+            <CircularLabel
+                label = { t('videoStatus.audioOnly') }
+                style = { normalizeStyleProps(_style, style) } />
+        );
+    }
+}
+
+/**
+ * Maps (parts of) the Redux state to the associated
+ * {@code VideoQualityLabel}'s props.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {{
+ *     _style: Object
+ * }}
+ */
+function _mapStateToProps(state: Object) {
+    const _abstractProps = _abstractMapStateToProps(state);
+
+    const _style
+        = _abstractProps._audioOnly ? styles.indicatorAudioOnly : undefined;
```

Comment:
I was following the discussion we had earlier about moving most of the calculation into the external function to avoid re-render, but I just realised that it doesn't make sense here as this property is not prone to frequent change, so I'll move it in, thanks!